api/src/main/java/org/openmrs/module/appointments/events/AppointmentEventType.java

```diff
@@ -0,0 +1,17 @@
+package org.openmrs.module.appointments.events;
+
+public enum AppointmentEventType {
+    BAHMNI_APPOINTMENT_CREATED("bahmni-appointment"),
+    BAHMNI_APPOINTMENT_UPDATED("bahmni-appointment"),
+    BAHMNI_RECURRING_APPOINTMENT_CREATED("bahmni-recurring-appointment"),
+    BAHMNI_RECURRING_APPOINTMENT_UPDATED("bahmni-recurring-appointment");
+
+    private final String topic;
+    AppointmentEventType(String topic) {
```

Comment:
we shouldn't be mixing event with the topic here. 
if any mapping needs to be done regarding which topic it must be sent to, then the listener to the spring application event (e.g. bahmni events module) - should use some mapping to identify the topic.

Reply:
@binduak ^